### ISSUE TITLE
Generation of outlets broken in Visual Studio For Mac 8.7.3 (build 13)

### ISSUE THREAD

**Issue by LuisAlbertoPenaNunez** [xamarin/xamarin-macios#9372]
When working in a Xamarin.iOS project creating outlets from xcode is broken currently only when the views subclass other classes.

### Steps to Reproduce

1. Create a view let's say ViewA.cs
2. Create a view let's say ViewB.cs (which subclass from UIView or anything UIView)
3. Make ViewA subclass ViewB (ViewA : ViewB)
4. In a ViewController with xib put a new View and set the custom class as ViewA.cs

### Expected Behavior

A new outlet should appear of type ViewA in ViewController.Designer.cs

### Actual Behavior

All outlets with custom classes turn into NSObjects

### Environment

```
=== Visual Studio Community 2019 for Mac ===

Version 8.7.3 (build 13)
Installation UUID: 04bc7f3d-05f0-4463-bc6f-01f09b0b654b
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 6.18.0.23 (d16-6 / 088c73638)

	Package version: 612000090

=== Mono Framework MDK ===

Runtime:
	Mono 6.12.0.90 (2020-02/d3daacdaa80) (64-bit)
	Package version: 612000090

=== Roslyn (Language Service) ===

3.7.0-6.20375.2+34202cc2f3e869fd70a26d8237f4552cf9e192cf

=== NuGet ===

Version: 5.7.0.6702

=== .NET Core SDK ===

SDK: /usr/local/share/dotnet/sdk/3.1.401/Sdks
SDK Versions:
	3.1.401
	3.1.302
	3.1.301
	3.1.300
	3.1.202
	3.1.201
	3.1.200
MSBuild SDKs: /Library/Frameworks/Mono.framework/Versions/6.12.0/lib/mono/msbuild/Current/bin/Sdks

=== .NET Core Runtime ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	3.1.7
	3.1.6
	3.1.5
	3.1.4
	3.1.3
	3.1.2
	2.1.21
	2.1.20
	2.1.19
	2.1.18
	2.1.17

=== Xamarin.Profiler ===

Version: 1.6.13.11
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Updater ===

Version: 11

=== Apple Developer Tools ===

Xcode 11.6 (16141)
Build 11E708

=== Xamarin.Mac ===

Xamarin.Mac not installed. Can't find /Library/Frameworks/Xamarin.Mac.framework/Versions/Current/Version.

=== Xamarin.iOS ===

Version: 13.20.2.2 (Visual Studio Community)
Hash: 817b6f72a
Branch: d16-7
Build date: 2020-07-18 18:45:00-0400

=== Xamarin Designer ===

Version: 16.7.0.492
Hash: f5afe667d
Branch: remotes/origin/d16-7-vsmac
Build date: 2020-07-10 18:42:54 UTC

=== Xamarin.Android ===

Version: 11.0.2.0 (Visual Studio Community)
Commit: xamarin-android/d16-7/025fde9
Android SDK: /Users/---/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		None installed

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 29.0.5
SDK Build Tools Version: 29.0.2

Build Information: 
Mono: 83105ba
Java.Interop: xamarin/java.interop/d16-7@1f3388a
ProGuard: Guardsquare/proguard/proguard6.2.2@ebe9000
SQLite: xamarin/sqlite/3.32.1@1a3276b
Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-7@017078f

=== Microsoft OpenJDK for Mobile ===

Java SDK: /Users/---/Library/Developer/Xamarin/jdk/microsoft_dist_openjdk_1.8.0.25
1.8.0-25
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Android SDK Manager ===

Version: 16.7.0.13
Hash: 8380518
Branch: remotes/origin/d16-7~2
Build date: 2020-08-13 18:19:24 UTC

=== Android Device Manager ===

Version: 16.7.0.22
Hash: 576004d
Branch: remotes/origin/d16-7
Build date: 2020-08-13 18:19:46 UTC

=== Build Information ===

Release ID: 807030013
Git revision: 793cad2e5999f9797bc6bc71759f31e97f57d94a
Build date: 2020-08-13 12:22:13-04
Build branch: release-8.7
Xamarin extensions: 793cad2e5999f9797bc6bc71759f31e97f57d94a

=== Operating System ===

Mac OS X 10.15.6
Darwin 19.6.0 Darwin Kernel Version 19.6.0
    Sun Jul  5 00:43:10 PDT 2020
    root:xnu-6153.141.1~9/RELEASE_X86_64 x86_64
```

### Build Logs

Visual studio doesn't seem to generate any when creating the outlets from objective-c to C# in the .designer.cs

### Sample project

[sample.zip](https://github.com/xamarin/xamarin-macios/files/5071541/sample.zip)

> VS bug [#1176884](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1176884)

**Comment by jeanpierrep1972** [comment]
I found something odd with this new VS Studio 2019 for Mac.. 

I don't know if this is a related issue. But I noticed that when I switch from Xcode to Xamarin studio. It recreates all the classes with the Register attribute. Even if the class already existed in another project or even elsewhere in the hierarchy.

Definitely what's broken with this version

**Comment by TesByRus** [comment]
Have same problem as TS and jeanpierrep1972 described. Very annoying.
Also in my projects .h and .m files generates at same folders where storyboard and xib files located. I think that all these problems are interconnected.

**Comment by jeanpierrep1972** [comment]
I think this is a bug from the new release VS 2019 Mac 8.7.2.. I rollbacked to preview version. It work fine.

I created a bug support. Hoping that will be fixed soon.

https://developercommunity.visualstudio.com/content/problem/1152754/syncing-from-xcode-when-back-into-vs-studio-2019.html

**Comment by whitneyschmidt** [comment]
FYI @maddyleger1

**Comment by projectgoav** [comment]
Effecting us as well. 

If you manaually edit the .designer files things seem quite happy but that isn't a long term fix :(

**Comment by jeanpierrep1972** [comment]
We have a new Visual Studio for Mac 8.7.3.13.. I wondering if it solve the problem.

**Comment by LuisAlbertoPenaNunez** [comment]
Nope @jeanpierrep1972 still broken

**Comment by jkapsouras** [comment]
No the problem is not solved. Very. annoying

**Comment by jkapsouras** [comment]
@jeanpierrep1972 problem remains in 8.7.4

**Comment by rafcabezas** [comment]
Is there a way to downgrade VS to 8.7.1?

**Comment by maddyleger1** [comment]
Hey folks! Xamarin PM here - we are still working on fixing this issue, but we're making progress. In the meantime you can download the last major VS Mac release (8.6) here: https://download.visualstudio.microsoft.com/download/pr/68ffa29a-6a5b-41f7-af7b-506ddcf4bbfc/6f601f3b5ff4d19ecd2c258517acc562/visualstudioformac-8.6.2.6.dmg 

We'll update you here when we have the fix in 8.7!

**Comment by BretJohnson** [comment]
As another temporary workaround, I believe if you move your view controller source files to be in the same directory as the .storyboard file, the problem won't happen.
If anyone else can confirm that, that would be good to know.

And, like maddy said, we're working on the real fix and will make it available as soon as we can.

**Comment by jkapsouras** [comment]
@BretJohnson I tried this workaround earlier today. It's ok if you don't have custom classes as outlets (eg. jkapisawsometextview :P ). If you have custom classes the synchronising with Xcode change them to NSObjects. Maybe you can do something like (varName as CustomClass).property but this is no solution if you have heavy controllers.

**Comment by jkapsouras** [comment]
I confirm @maddyleger1 's comment! 8.6 works perfect. The good old two Xcode windows. No new files, no broken outlets! Thanks for the download link, you saved my day

**Comment by pkhandelwal-cdl** [comment]
Another workaround is to do all the layout stuff using XCode and then define the outlets and custom classes using the built-in designer. I am using individual XIB files.

**Comment by jkapsouras** [comment]
Update from developer community of visual studio: A fix for this issue has been internally implemented and is being prepared for release. We’ll update you once it becomes available for download.

**Comment by jkapsouras** [comment]
8.7.5.19 released (https://docs.microsoft.com/en-us/visualstudio/releasenotes/vs2019-mac-relnotes#875) and the problem seems solved. Just updated, edited a storyboard and new files were created. :) I also created an outlet from a custom control and it's working perfectly. I think that this issue can be marked closed along with #9583

**Comment by whitneyschmidt** [comment]
Closing this issue as we have verified here and in https://github.com/xamarin/xamarin-macios/issues/9583 that is fixed in 8.7.5.19!
